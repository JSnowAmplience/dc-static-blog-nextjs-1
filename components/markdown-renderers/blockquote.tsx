import theme from '../../common/styles/default/theme';

interface MarkdownBlockquoteProps {
  children: JSX.Element;
}

export const MarkdownBlockquote = ({ children }: MarkdownBlockquoteProps) => {
  return (
    <>
      <blockquote>{children}</blockquote>
      <style jsx>{`
        blockquote {
          color: ${theme.colors.mineShaft};
          padding: 10px;
          border-left: 2px solid ${theme.colors.radicalRed};
          margin: 25px;
        }

        blockquote :global(p) {
          color: ${theme.colors.mineShaft};
          margin: 0;
          font-size: 1.3125rem;
          font-weight: ${theme.fonts.weight.regular};
          line-height: 2rem;
          padding: 6px;
        }

        @media (max-width: ${theme.layout.narrowPageWidth}) {
          blockquote :global(p) {
            line-height: 3.125rem;
          }
        }
      `}</style>
    </>
  );
};
