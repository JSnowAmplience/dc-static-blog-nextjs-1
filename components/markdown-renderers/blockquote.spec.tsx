import renderer from 'react-test-renderer';
import { MarkdownBlockquote } from './blockquote';

describe('MarkdownBlockquote', (): void => {
  it('should render some code as a pre tag with syntax highlighting', (): void => {
    const props = {
      children: <>test text</>
    };

    const wrapper = renderer.create(<MarkdownBlockquote {...props} />).toJSON();
    expect(wrapper).toMatchSnapshot();
  });
});
