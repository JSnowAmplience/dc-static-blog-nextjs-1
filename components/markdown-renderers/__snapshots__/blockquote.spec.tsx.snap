// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownBlockquote should render as a blockquote 1`] = `
<blockquote
  className="jsx-6158228"
>
  test text
</blockquote>
`;
